subsys/pm/device_runtime.c

```diff
@@ -172,12 +174,12 @@ void pm_device_runtime_enable(const struct device *dev)
 
 	SYS_PORT_TRACING_FUNC_ENTER(pm, device_runtime_enable, dev);
 
-	if (!k_is_pre_kernel()) {
-		(void)k_mutex_lock(&pm->lock, K_FOREVER);
+	if (atomic_test_bit(&pm->flags, PM_DEVICE_FLAG_RUNTIME_ENABLED)) {
```

Comment:
note: this change breaks `pm_device_runtime_enable` thread safety (even though consequences in this case are not bad, but it could be if enable gets more code added and user is not aware)